Automattic/wp-calypso
# Jetpack Cloud: add support for highlighted products on Pricing page

#### Changes proposed in this Pull Request

* Add support to highlight product cards on the Pricing page. Products can be highlighted by adding to the URL the query parameter `plan=[product slug]`.

#### Implementation notes

We don't know, from a design perspective, how cards will eventually be highlighted. For this reason, I decided to only implement a behavior that scrolls the highlighted card into view. This can be easily extended to accommodate any other type of visual effect or behavior.

To make testing easier, I added a query parameter called `plan` that one can use to select which product should be highlighted.

#### Testing instructions

* Download this PR.
* Run `yarn start-jetpack-cloud` to start Jetpack Green.
* Visit the pricing page located at `/pricing`.
* Add the following query parameter to the URL: `?plan=jetpack_backup_daily`.
* Make sure that Jetpack Backup Daily was scrolled into view.
* Add the following query parameter to the URL: `?plan=jetpack_search`.
* Make sure that Jetpack Search was scrolled into view.
* Add the following query parameter to the URL: `?plan=jetpack_complete_monthly`.
* Make sure that Jetpack Complete was scrolled into view.
* Add the following query parameter to the URL: `?plan=jetpack_scan`.
* Make sure that Jetpack Scan was scrolled into view.
* Repeat the same steps on a smaller viewport (one and two columns layout).
* Make sure that always the right card appears and that is not hiding behind the filter bar.

Related to 1200196139286276-as-1200233771803638

#### Demo

##### 3-columns layout
https://user-images.githubusercontent.com/3418513/115729423-c822b000-a35b-11eb-83e7-3c4cb9837755.mp4

##### 1-column layout
https://user-images.githubusercontent.com/3418513/115729434-c9ec7380-a35b-11eb-8638-fbf935ce35da.mp4
